api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -0,0 +1,162 @@
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE InstanceSigs #-}
+
+module OpenTelemetry.Internal.Logging.Types (
+  Logger (..),
+  LogRecord (..),
+  LogRecordArguments (..),
+  mkSeverityNumber,
+  shortName,
+  severityInt,
+) where
+
+import Data.Int (Int64)
+import Data.Text (Text)
+import OpenTelemetry.Common (Timestamp, TraceFlags)
+import OpenTelemetry.Context.Types
+import OpenTelemetry.Internal.Trace.Id (SpanId, TraceId)
+import OpenTelemetry.LogAttributes (LogAttributes)
+import OpenTelemetry.Resource (MaterializedResources)
+
+
+-- | LogRecords can be Created from Loggers
+data Logger = Logger
+  { loggerResource :: Maybe MaterializedResources
+  }
```

Comment:
`Logger` will likely expand to more fields as necessary, but `Resource` is the only one needed at the moment.